operations/restrictions.md

```diff
@@ -232,10 +232,10 @@ Adds new restrictions with the specified conditions. Care is needed to specify `
 | Property | Type | Contract | Description |
 | :-- | :-- | :-- | :-- |
 | `Type` | [Restriction type](restrictions.md#restriction-type) | required | Restriction type. |
-| `ExactRateId` | string | optional | Unique identifier of the exact `Rate` to which the restriction applies. |
-| `BaseRateId` | string | optional | Unique identifier of the base `Rate` to which the restriction applies. |
-| `RateGroupId` | string | optional | Unique identifier of the `Rate group` to which the restriction applies. |
-| `ResourceCategoryId` | string | optional | Unique identifier of the `Resource category` to which the restriction applies. |
+| `ExactRateId` | [Hybrid identifier](_objects.md#hybrid-identifier) | optional | Unique identifier of the exact `Rate` to which the restriction applies. |
```

Comment:
Not sure if hybrid identifier should be displayed as a type. Looks like a breaking change at the firstglance, and for the majority of API users the hybrid is not relevant. An alternative is to keep it as a string, but modify the description. It sure is neater to implement as a type though.

Reply:
Imho this is a better approach, because you see more easily what supports it's or not and not looking into description. Introduction of Hybrid Identifier should be in the changelog (didn't put it there yet, until we agree on the rest) mentioning it's not a breaking change.

Reply:
Another thing to consider - will Hybrid identifier always mean that it is Id + ExternalIdentifier? It seems unlikely, but if we have a different combination of properties we will have to separate them in the documentation.

Reply:
Yes but I would leave that for future imho 🤔

Reply:
This is an interesting case! My first thought is to agree with @jonas-paul, it looks like we are changing the **type** from `string` to a custom `object`, which is not the case. I suppose it depends on what is meant by **type** in the table? In my opinion, **type** should be the JSON type, so it should _always_ be either `string`, `number`, `integer`, `object`, `boolean`, `decimal`, or `array of <data type>`. We could then add any special qualifiers afterwards, e.g. `object <link to object definition>` or `string <link to string enumeration>`.

I think the same logic would apply here, so my personal recommendation would be: `string [Hybrid identifier](_objects.md#hybrid-identifier)`. Then it is clear that this is a `string`, but there is also a link to show what values are permitted for the string, just like a string enumeration.

Regarding backward compatibility, this would depend on what the constraints are for `Hybrid identifier`, but in this case I can see it is fully backwards compatible, so it would be good just to make that 100% clear. I think it might be good to add a sentence to that effect in the description for `Hybrid identifier` and also in the Changelog entry.

Reply:
Just to summarise @VBuben:
1. I recommend `string [Hybrid identifier](_objects.md#hybrid-identifier)` for the data type.
2. I recommend adding a sentence to the Hybrid identifier description (in [common objects](_objects.md)) to say that it is entirely back compatible with simply using the usual string identifier.
3.  I recommend adding a sentence to the Changelog to say that the change is backwards compatible and NOT a breaking change.

Reply:
I agree with Mike. We can modify the generator to add string prefix; I'd probably scope it to `hybrididentifier` for now so it doesn't affect all other properties.

Reply:
Modified generator and regenerated in [`519953b` (#661)](https://github.com/MewsSystems/gitbook-connector-api/pull/661/commits/519953b8842453e14d4b5b9c1bd10bd3a6cc526e)